src/Pact/Types/ExpParser.hs

```diff
@@ -136,12 +137,13 @@ runCompile act cs a =
     (Left (TrivialError _ (Just err) expect)) -> case err of
       EndOfInput -> case S.toList expect of
         (Tokens (x :| _):_) -> doErr (getInfo x) "unexpected end of input"
-        (Label s:_) -> doErr def (toList s)
-        er -> doErr def (show er)
-      Label ne -> doErr def (toList ne)
-      Tokens (x :| _) -> doErr (getInfo x) $ showExpect expect
-    (Left e) -> doErr def (show e)
-    where doErr i s = Left $ PactError SyntaxError i def (pack s)
+        (Label s:_) -> doErr def (text (toList s))
```

Comment:
This is such a mess, but I guess that's left for another PR. I'm pretty sure there's a bunch of errors with terrible context atm.